core/aws_ddk_core/resources/_kinesisfirehose.py

```diff
@@ -0,0 +1,172 @@
+            Assumed by Kinesis Data Firehose to read from sources and encrypt data server-side.
+            Default: - a role will be created with default permissions.
+        source_stream: Optional[IStream] = None
+            The Kinesis data stream to use as a source for this delivery stream
+        **firehose_props: Any
+            Additional properties. For complete list of properties refer to CDK Documentation -
+            Firehose Delivery Stream:
+            https://docs.aws.amazon.com/cdk/api/v1/python/aws_cdk.aws_kinesisfirehose/DeliveryStream.html
+
+        Returns
+        -------
+        delivery_stream: firehose.IDeliveryStream
+            A Kinesis Firehose Delivery Stream
+        """
+        # Collect args
+        firehose_props = {
+            "delivery_stream_name": delivery_stream_name,
+            "destinations": destinations,
+            "encryption": encryption,
+            "encryption_key": encryption_key,
+            "role": role,
+            "source_stream": source_stream,
+            **firehose_props,
+        }
+
+        # create delivery stream
+        _logger.debug(f"firehose_props: {firehose_props}")
+        firehose_stream: firehose.IDeliveryStream = firehose.DeliveryStream(scope, id, **firehose_props)
+
+        return firehose_stream
+
+    @staticmethod
+    def s3_destination(
+        environment_id: str,
+        id: str,
+        bucket: IBucket,
+        buffer_interval: Optional[Duration] = None,
+        buffer_size: Optional[Size] = None,
+        **destination_props: Any,
+    ) -> firehose.IDeliveryStream:
+        """
+        Create and configure Firehose delivery S3 destination.
+
+        This construct allows to configure parameters of the firehose destination using ddk.json
+        configuration file depending on the `environment_id` in which the function is used.
+        Supported parameters are:
+
+        Parameters
+        ----------
+        id : str
+            Identifier of the queue
+        environment_id : str
+            Identifier of the environment
+        **destination_props: Any
+            Additional properties. For complete list of properties refer to CDK Documentation -
+            Firehose S3 Destinations:
+            https://docs.aws.amazon.com/cdk/api/v1/python/aws_cdk.aws_kinesisfirehose_destinations/S3Bucket.html
+
+        Returns
+        -------
+        destination: destinations.S3Bucket
+            A Kinesis Firehose S3 Delivery Destination
+        """
+        # Load and validate the config
+        destination_config_props: Dict[str, Any] = FirehoseDestinationSchema().load(
+            Config().get_resource_config(
+                environment_id=environment_id,
+                id=id,
+            ),
+            partial=["removal_policy"],
+        )
+
+        # Collect args
+        destination_props = {
+            "buffer_interval": buffer_interval,
+            "buffer_size": buffer_size,
+            **destination_props,
```

Comment:
Would we want to set GZIP as a sane default over no compression? What do you think @kukushking?
https://docs.aws.amazon.com/cdk/api/v2/python/aws_cdk.aws_kinesisfirehose_destinations_alpha/Compression.html#aws_cdk.aws_kinesisfirehose_destinations_alpha.Compression